api/src/test/java/org/cardanofoundation/rosetta/api/construction/service/CardanoConstructionServiceImplTest.java

```diff
@@ -94,6 +98,26 @@ void outputMoreThanInputTest() {
     assertEquals("The transaction you are trying to build has more outputs than inputs", exception.getError().getMessage());
   }
 
+  @Test
+  void calculateTxSize_whenUnsignedTransactionAddressesInvalid_thenThrowException() {
+    List<Operation> operations = Collections.singletonList(givenOperation());
+    try (MockedStatic<CardanoAddressUtils> mockedAddressUtil = Mockito.mockStatic(
+        CardanoAddressUtils.class)) {
+      mockedAddressUtil.when(() -> CardanoAddressUtils.getEraAddressType(anyString()))
+          .thenReturn(null);
+      mockedAddressUtil.when(() -> CardanoAddressUtils.isEd25519KeyHash(anyString()))
+          .thenReturn(false);
+
+      ApiException result = assertThrows(ApiException.class,
+          () -> cardanoService.calculateTxSize(NetworkIdentifierType.CARDANO_TESTNET_NETWORK,
+              operations, 0, null));
+
+      assertEquals(RosettaErrorType.INVALID_ADDRESS.getMessage(), result.getError().getMessage());
+      assertEquals(RosettaErrorType.INVALID_ADDRESS.getCode(), result.getError().getCode());
+      assertTrue(result.getError().isRetriable());
```

Comment:
It's a good practice to check service interactions to make the flow explicit

Reply:
done